saleor/graphql/product/tests/test_variant.py

```diff
@@ -4674,3 +4674,73 @@ def test_query_product_variant_for_federation(api_client, variant, channel_USD):
             "name": variant.name,
         }
     ]
+
+
+VARIANT_CREATE_MUTATION = """
+    mutation variantCreate($productId: ID!, $attrId: ID!) {
+        productVariantCreate (
+            input: {
+                sku: "my-sku",
+                product: $productId,
+                attributes: [{id: $attrId, values:["1"]}]
+            }
+        )
+        {
+            productVariant {
+                id
+            }
+            errors {
+                field,
+                message,
+            }
+        }
+    }
+"""
+
+
+def test_variant_create_product_without_variant_attributes(
+    product_with_product_attributes, staff_api_client, permission_manage_products
+):
+    product = product_with_product_attributes
+
+    prod_id = graphene.Node.to_global_id("Product", product.pk)
+    attr_id = graphene.Node.to_global_id(
+        "Attribute", product.product_type.product_attributes.first().pk
+    )
+
+    response = staff_api_client.post_graphql(
+        VARIANT_CREATE_MUTATION,
+        variables={"productId": prod_id, "attrId": attr_id},
+        permissions=[permission_manage_products],
+    )
+    content = get_graphql_content(response)
+
+    errors = content["data"]["productVariantCreate"]["errors"]
+    assert errors
+    assert errors[0]["message"] == "Given attribute is not a variant attribute"
```

Comment:
We usually do not check the error messages, we are checking `code` and `field` of an error. Checking messages is connected with fixing every test that checks this, even if a dot is added.